sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1207,6 +1207,13 @@ object SQLConf {
       .intConf
       .createWithDefault(100)
 
+  val RANGE_EXCHANGE_SAMPLE_CACHE_ENABLE =
+    buildConf("spark.sql.execution.rangeExchange.sampleCache.enabled")
+      .doc("When true, if the sampled data in range partitioning covers the whole data, " +
+        "the sampled data will be cached for the next sort.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
Should it be an internal config?

Reply:
I am also somewhat uncertain. What do you think of it?

Reply:
I am also somewhat uncertain. What do you think of it?
@kiszk

Reply:
Do we expect users to tune it?